test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -1164,6 +1183,38 @@ UCS_TEST_P(test_ucp_sockaddr_destroy_ep_on_err, onesided_bidi_sforce) {
     one_sided_disconnect(sender(),   UCP_EP_CLOSE_MODE_FLUSH);
 }
 
+UCS_TEST_P(test_ucp_sockaddr_destroy_ep_on_err, create_and_destroy_immediately)
+{
+    ucp_test_base::entity::listen_cb_type_t listen_cb_type = cb_type();
+
+    listen(listen_cb_type);
+
+    {
+        scoped_log_handler warn_slh(detect_warn_logger);
+        scoped_log_handler error_slh(detect_error_logger);
+        client_ep_connect();
+
+        if (listen_cb_type == ucp_test_base::entity::LISTEN_CB_CONN) {
+            while ((m_err_count == 0) &&
+                   receiver().is_conn_reqs_queue_empty()) {
+                progress();
+            }
+        } else {
+            ASSERT_EQ(ucp_test_base::entity::LISTEN_CB_EP, listen_cb_type);
+            if (!wait_for_server_ep(false)) {
+                UCS_TEST_SKIP_R("cannot connect to server");
+            }
+        }
+
+        one_sided_disconnect(sender(), UCP_EP_CLOSE_MODE_FORCE);
+        while ((m_err_count == 0) && (receiver().get_err_num() == 0)) {
```

Comment:
should we add a timeout here? to prevent a hang in case of a failure.

Reply:
good catch, done